### PR TITLE
WIP: big update

### PR DESCRIPTION
**Pull request by cgibbard**
This merges cg-rhyolite-vessel with a bunch of changes from @eskimor and others that were needed for a particular project. It should be reviewed after cg-rhyolite-vessel is merged.

### CODE REVIEW

**Comment by ali-abrar** [comment]
Changelog!

**Comment by cgibbard** [comment]
Yep, I'll make sure it has one before we actually merge

File: ChangeLog.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ This project's release branch is `master`. This log is written from the perspect
 
 ## 2019-06-14 - Unreleased
 
+* Remove the "HasView" and "HasRequest" classes, and the general concept of having a type level "app" identifier. Instead, everything is explicitly parametrised on query and request types directly, and the query type is no longer *required* to be a Functor/Align/etc. so that Vessel becomes an option for defining queries and views.
```
Comment (3noch) [inline ChangeLog.md]: parameterized (note this is more than just American vs English spelling)

Comment (cgibbard) [reply ChangeLog.md]: I... I think they're both right?

Comment (3noch) [reply ChangeLog.md]: Hm, you're right. It's an alternative spelling that seems to be less popular or something... TIL

**Comment by ali-abrar** [comment]
Is this still WIP? Does it include #72 and #70 ?

**Pull request closed without merging**